CMakeLists.txt

```diff
@@ -0,0 +1,845 @@
+    else()
+      string(REPLACE "/" "\\" groupname ${curdir})
+      source_group(${groupname} FILES
+        ${PROJECT_SOURCE_DIR}/${curdir}/${child})
+    endif()
+  endforeach()
+endmacro()
+
+macro(add_with_props files)
+  list(APPEND src ${files})
+  foreach (arg ${ARGN})
+    set(props "${props} ${arg}")
+  endforeach()
+  set_source_files_properties(
+    ${files}
+    PROPERTIES COMPILE_FLAGS
+    ${props})
+endmacro()
+
+############################################################
+
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES ".*Clang") # both Clang and AppleClang
+  set(is_clang true)
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+  set(is_gcc true)
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+  set(is_msvc true)
+endif()
+
+if (${CMAKE_GENERATOR} STREQUAL "Xcode")
+  set(is_xcode true)
+else()
+  set(is_xcode false)
+endif()
+
+if (NOT is_gcc AND NOT is_clang AND NOT is_msvc)
+  message("Current compiler is ${CMAKE_CXX_COMPILER_ID}")
+  message(FATAL_ERROR "Missing compiler. Must be GNU, Clang, or MSVC")
+endif()
+
+############################################################
+# Check if should use gcc4's ABI
+set(gcc4_abi false)
+
+if ($ENV{RIPPLED_OLD_GCC_ABI})
+  set(gcc4_abi true)
+endif()
+
+if (is_gcc AND NOT gcc4_abi)
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5)
+    execute_process(COMMAND lsb_release -si OUTPUT_VARIABLE lsb)
+    string(STRIP ${lsb} lsb)
+    if (${lsb} STREQUAL "Ubuntu")
+      execute_process(COMMAND lsb_release -sr OUTPUT_VARIABLE lsb)
+      string(STRIP ${lsb} lsb)
+      if (${lsb} VERSION_LESS 15.1)
+        set(gcc4_abi true)
+      endif()
+    endif()
+  endif()
+endif()
+
+if (gcc4_abi)
+  add_definitions(-D_GLIBCXX_USE_CXX11_ABI=0)
+endif()
+
+############################################################
+
+include_directories(src src/beast src/beast/include src/beast/extras
+  src/soci/src src/soci/include)
+
+if (coverage)
+  add_compile_options(-fprofile-arcs -ftest-coverage)
+  append_flags(CMAKE_EXE_LINKER_FLAGS -fprofile-arcs -ftest-coverage)
+endif()
+
+if (profile)
+  add_compile_options(-p -pg)
+  append_flags(CMAKE_EXE_LINKER_FLAGS -p -pg)
+endif()
+
+############################################################
+
+if (NOT WIN32)
+  if (is_clang AND DEFINED ENV{CLANG_BOOST_ROOT})
+    set(BOOST_ROOT $ENV{CLANG_BOOST_ROOT})
+  endif()
+
+
+  set(Boost_USE_STATIC_LIBS on)
+  set(Boost_USE_MULTITHREADED on)
+  set(Boost_USE_STATIC_RUNTIME off)
+  find_package(Boost COMPONENTS
+    coroutine
+    context
+    date_time
+    filesystem
+    program_options
+    regex
+    system
+    thread)
+
+  if (Boost_FOUND)
+    include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
+  else()
+    message(FATAL_ERROR "Boost not found")
+  endif()
+endif()
+
+if (APPLE)
+  # swd TBD fixme
+  file(GLOB OPENSSL_ROOT_DIR /usr/local/Cellar/openssl/*)
+  # set(OPENSSL_ROOT_DIR /usr/local/Cellar/openssl)
+endif()
+
+if (WIN32)
+  if (DEFINED ENV{OPENSSL_ROOT})
```

Comment:
seems like the OpenSSL find module should still work on windows...just have to set OPENSSL_ROOT_DIR to OPENSSL_ROOT always for win, since there are no system lib/inc dirs